kornia/color/adjust.py

```diff
@@ -1,42 +1,244 @@
+
+
+def adjust_contrast(input: torch.Tensor,
+                    contrast_factor: Union[float, torch.Tensor]) -> torch.Tensor:
+    r"""Adjust Contrast of an image.
+
+    See :class:`~kornia.color.AdjustContrast` for details.
+    """
+
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(contrast_factor, (float, torch.Tensor,)):
+        raise TypeError(f"The factor should be either a float or torch.Tensor. "
+                        f"Got {type(contrast_factor)}")
+
+    if isinstance(contrast_factor, float):
+        contrast_factor = torch.tensor([contrast_factor])
+
+    contrast_factor = contrast_factor.to(input.device).to(input.dtype)
+
+    if (contrast_factor < 0).any():
+        raise ValueError(f"Contrast factor must be non-negative. Got {contrast_factor}")
+
+    for _ in input.shape[1:]:
+        contrast_factor = torch.unsqueeze(contrast_factor, dim=-1)
+
+    # Apply brightness factor to each channel
+    x_adjust: torch.Tensor = input + contrast_factor
+
+    # Truncate between pixel values
+    out: torch.Tensor = torch.clamp(x_adjust, 0.0, 1.0)
+
+    return out
+
+
+def adjust_brightness(input: torch.Tensor,
+                      brightness_factor: Union[float, torch.Tensor]) -> torch.Tensor:
+    r"""Adjust Brightness of an image.
 
     See :class:`~kornia.color.AdjustBrightness` for details.
+    """
+
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(brightness_factor, (float, torch.Tensor,)):
+        raise TypeError(f"The factor should be either a float or torch.Tensor. "
+                        f"Got {type(brightness_factor)}")
+
+    if isinstance(brightness_factor, float):
+        brightness_factor = torch.tensor([brightness_factor])
+
+    brightness_factor = brightness_factor.to(input.device).to(input.dtype)
+
+    if (brightness_factor < 0).any():
+        raise ValueError(f"Brightness factor must be non-negative. Got {brightness_factor}")
+
+    for _ in input.shape[1:]:
+        brightness_factor = torch.unsqueeze(brightness_factor, dim=-1)
+
+    # Apply brightness factor to each channel
+    x_adjust: torch.Tensor = input * brightness_factor
+
+    # Truncate between pixel values
+    out: torch.Tensor = torch.clamp(x_adjust, 0.0, 1.0)
+
+    return out
+
+
+class AdjustSaturation(nn.Module):
+    r"""Adjust color saturation of an image.
+
+    The input image is expected to be an RGB image in the range of [0, 1].
 
     Args:
-        image (torch.Tensor): Image to be adjusted.
-        brightness_factor (torch.Tensor): Brightness adjust factor per element
-          in the batch. 0 generates a compleatly black image, 1 does not modify
-          the input image while any other non-negative number modify the
-          brightness by this factor.
+        image (torch.Tensor): Image to be adjusted in the shape of (*, N).
```

Comment:
Don`t get the shape, actually. If it is (*, N), than it is not an image, but any nd tensor, n>1

Reply:
I got feedback from users that might want to apply data augmentation on flatten images :D

Reply:
I mean, just rename Image to Image/Tensor in docstring to avoid confusion

Reply:
ah sure